### PR TITLE
Implement a simpler toHiveString to convert spark row

### PR DESCRIPTION
**Pull request by yaooqinn**
yaooqinn [Closes #400](https://github.com/yaooqinn/kyuubi/pull/400)    Target Issue Bug [&#10088;?&#10089;](https://pullrequestbadge.com/?utm_medium=github&utm_source=yaooqinn&utm_campaign=badge_info)

### _Why are the changes needed?_

Hive compatibility is essential! This PR tries to restore the previous behavior.

1. Not use Spark's internal class HiveResult to convert spark catalyst type-values to JDBC results, the interval APIs are not APIs and changed unexpectedly. It involves a lot of overhead. (For this reason, @pan3793 and I propose a Casting way to solve it.)

2. Unfortunately, the casting way breaks things like #397. So this is a reverting PR.

3. I checked the spark `HiveResult` carefully, I noticed that the `nested` logic is not needed in Kyuubi as we already matched primitives ahead.
4. Also the date time formatter used in `HiveResult` is excessively packaged with massively complicated logic inside spark. It's not clear and simple enough to call.
5. (tiny breaking change)the trailing spaces of a timestamp in milliseconds may not be trimmed. But this is a correctness issue and acceptable.
 

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
The Projection-Cast way seems to introduce a regression of https://github.com/yaooqinn/kyuubi/issues/397, cc @pan3793 

In this PR, I re-implement the `toHiveString` partially from Spark

also cc @Honglun, can you help to verify this and thanks for the bug report

**Comment by Honglun** [comment]
I don't understand the purpose of this code( from line 69 to 78). Why don't we just use 'result' like this 
'iter = new ArrayFetchIterator(result.collect())' at line 80 ?

**Comment by Honglun** [comment]
Missing the file I mentioned above:
https://github.com/yaooqinn/kyuubi/blob/master/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala#L78

**Comment by yaooqinn** [comment]
> I don't understand the purpose of this code( from line 69 to 78). Why don't we just use 'result' like this
> 'iter = new ArrayFetchIterator(result.collect())' at line 80 ?

the deleted code  is used for converting spark value to JDBC value

File: kyuubi-common/src/test/scala/org/apache/kyuubi/operation/JDBCTests.scala [hunk 1]
```diff
@@ -215,8 +215,8 @@ trait JDBCTests extends BasicJDBCTests {
       val resultSet = statement.executeQuery(
         "SELECT struct('1', '2') AS col1, named_struct('a', 2, 'b', 4) AS col2")
       assert(resultSet.next())
-      assert(resultSet.getObject("col1") === "[1, 2]")
-      assert(resultSet.getObject("col2") === "[2, 4]")
+      assert(resultSet.getObject("col1") === "{\"col1\":\"1\",\"col2\":\"2\"}")
+      assert(resultSet.getObject("col2") === "{\"a\":2,\"b\":4}")
```
Comment (pan3793) [inline kyuubi-common/src/test/scala/org/apache/kyuubi/operation/JDBCTests.scala:219]: use """ to avoid back slash

**Review by pan3793 — APPROVED** [review]

**Comment by yaooqinn** [comment]
Hi, @Honglun, can you tell us whether the patch works for you or not? If yes, can you make a follow-up PR to add your test case to the master branch to ensure we don't break anything related again?

also cc @ulysses-you @zwangsheng

**Comment by Honglun** [comment]
> Hi, @Honglun, can you tell us whether the patch works for you or not? If yes, can you make a follow-up PR to add your test case to the master branch to ensure we don't break anything related again?
> 
> also cc @ulysses-you @zwangsheng

Is this patch already available for branch '397' ?

**Comment by yaooqinn** [comment]
> > Hi, @Honglun, can you tell us whether the patch works for you or not? If yes, can you make a follow-up PR to add your test case to the master branch to ensure we don't break anything related again?
> > also cc @ulysses-you @zwangsheng
> 
> Is this patch already available for branch '397' ?

yes, I'd like you to help check your case locally first or I can just merge it to master then you start your investigation?

**Comment by Honglun** [comment]
> > > Hi, @Honglun, can you tell us whether the patch works for you or not? If yes, can you make a follow-up PR to add your test case to the master branch to ensure we don't break anything related again?
> > > also cc @ulysses-you @zwangsheng
> > 
> > 
> > Is this patch already available for branch '397' ?
> 
> yes, I'd like you to help check your case locally first or I can just merge it to master then you start your investigation?

OK, Let me test branch 397 first.

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/schema/RowSet.scala [hunk 2]
```diff
@@ -164,19 +191,90 @@ object RowSet {
...
+   * A simpler impl of Spark's toHiveString
+   */
+  def toHiveString(dataWithType: (Any, DataType), timeZone: ZoneId): String = {
+    dataWithType match {
+      case (null, _) =>
+        // Only match nulls in nested type values
+        "null"
+
+      case (d: Date, DateType) =>
+        simpleDateFormat.format(d)
```
Comment (ulysses-you) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/schema/RowSet.scala:242]: nit: `dateFormatter.format(d)`

**Comment by ulysses-you** [comment]
If I understand correctly, this PR try to fix two problem ?
1. result which is not alias can not be cast. 
2. timezone not respect the config `spark.sql.session.timeZone`.

**Comment by yaooqinn** [comment]
> If I understand correctly, this PR try to fix two problem ?
> 
> 1. result which is not alias can not be cast.
> 2. timezone not respect the config `spark.sql.session.timeZone`.

I updated the PR description a bit for you guys to understand and review

**Comment by Honglun** [comment]
I tested 'select function' locally, and it is normal.

**Comment by yaooqinn** [comment]
> I tested 'select function' locally, and it is normal.

can you submit a PR with your test case later, and thanks for the help

**Comment by Honglun** [comment]
> > I tested 'select function' locally, and it is normal.
> 
> can you submit a PR with your test case later, and thanks for the help

OK, I'll submit it later.

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master for 1.1.0
